cpp/src/strings/regex/regex_program.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "regex_program_impl.h"
+
+#include <cudf/strings/regex/regex_program.hpp>
+
+#include <memory>
+#include <string>
+
+namespace cudf {
+namespace strings {
+
+std::unique_ptr<regex_program> regex_program::create(std::string_view pattern,
+                                                     regex_flags flags,
+                                                     capture_groups capture)
+{
+  auto p = new regex_program(pattern, flags, capture);
+  return std::unique_ptr<regex_program>(p);
```

Comment:
Just checking: we can't use `make_unique` here because the constructor is private, right?

Reply:
Correct.